docs/testplan/subnet_decap/Subnet-decap-test-plan.md

```diff
@@ -0,0 +1,181 @@
+  - Inner SIP: Any IP
+
+2. Ingress the packet from upstream ports to DUT.
+3. Verify packet is decapsulated and routed to inner DIP.
+
+#### IPV4 negative scenario
+1. Craft `IPinIP` packet with signature as below.
+
+  - Outer DIP: Any IP in `192.168.0.1/21`
+  - Outer SIP: Any IP **NOT** in `20.20.20.0/24`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+2. Setup ARP responder on ptf and trigger MAC learning with `ping` from DUT.
+3. Ingress the packet from upstream ports to DUT.
+4. Verify packet is forwarded to expected VLAN port.
+
+#### IPV6 positive scenario
+1. Craft `IPV6inIPV6` packet with signature as below.
+
+  - Outer DIP: Any IP in `fc02:1000::1/64`
+  - Outer SIP: Any IP in `fc01::0/120`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+
+2. Ingress the packet from upstream ports to DUT.
+3. Verify packet is decapsulated and routed to inner DIP.
+
+#### IPV6 negative scenario
+1. Craft `IPV6inIPV6` packet with signature as below.
+
+  - Outer DIP: Any IP in `fc02:1000::1/64`
+  - Outer SIP: Any IP **NOT** in `fc01::0/120`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+
+2. Setup ARP responder on ptf and trigger NBR discovery with `ping` from DUT.
+3. Ingress the packet from upstream ports to DUT.
+4. Verify packet is forwarded to expected VLAN port.
+
+### Test case 2 - Verify IPinIP packet targeting at VIP is decapsulated (T0 only)
+#### Test setup
+1. Setup SLB BGP neighbor on ptf and establish BGP sessions with DUT.
+2. Announce some routes (including both IPv4 and IPv6) to DUT from SLB.
+#### Test steps
+#### IPV4 scenario
+1. Verify `STATE_DB` entries are created as expected.
+2. Craft packet with below signature
+  - Outer DIP: Any IP in the prefix announced by SLB
+  - Outer SIP: Any IP in `20.20.20.0/24`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+3. Ingress the packet from upstream ports to DUT.
+4. Verify packet is decapsulated and forwarded to inner_dip.
+5. Withdraw IPv4 prefix announced from SLB.
+6. Verify `STATE_DB` entries are removed as expected.
+7. Ingress the same packet and verify decapsulation is not happening.
+
+#### IPV6 scenario
+1. Verify `STATE_DB` entries are created as expected.
+2. Craft packet with below signature
+  - Outer DIP: Any IP in the prefix announced by SLB
+  - Outer SIP: Any IP in `fc01::0/120`
+  - Inner DIP: Any dummy IP routed by default route
+  - Inner SIP: Any IP
+3. Ingress the packet from upstream ports to DUT.
+4. Verify packet is decapsulated and forwarded to inner_dip.
+5. Withdraw IPv6 prefix announced from SLB.
+6. Verify `STATE_DB` entries are removed as expected.
+7. Ingress the same packet and verify decapsulation is not happening.
+
+### Test case 3 - Verify IPinIP packet targeting at vnet route is decapsulated (T1 only)
+#### Test steps
+#### IPV4 scenario
+1. Configure IPv4 vnet routes with RESTAPI on DUT.
```

Comment:
Better to be more specific on the steps:
1. add vxlan tunnel
2. add vnet
3. add vxlan tunnel route

Reply:
Fixed.